lib/KCard/BaseCard.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+
+  <li
+    :class="['remove-list-style card',$computedClass(coreOutlineFocus)]"
+    tabindex="0"
+    data-focus="true"
+    :style="cardStyle"
+    @focus="cardFocus"
+    @mouseenter="cardHover"
+    @click="cardClickHandler"
+    @mousedown="onMouseDown"
+    @mouseup="onMouseUp"
+    @keyup.enter="cardClick"
+  >
+    <component
+      :is="headerLevel"
+      v-if="title || $slots.title"
+      class="base-card-heading"
+    >
+      <router-link
+        tabindex="-1"
+        :to="to"
+      >
+        <slot v-if="$slots.title" name="title"></slot>
+        <KTextTruncator
+          v-else
+          :text="title"
+          :maxLines="titleLines"
+          :style="titleStyle"
+        />
+      </router-link>
+    </component>
+
+    <slot name="default"></slot>
+  </li>
+
+</template>
+
+
+<script>
+
+  export default {
+    name: 'BaseCard',
+
+    props: {
+      to: {
+        type: Object,
+        required: true,
+      },
+      title: {
+        type: String,
+        required: false,
+        default: null,
+      },
+      titleLines: {
+        type: Number,
+        required: false,
+        default: 2,
+      },
+      headingLevel: {
+        type: Number,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        mouseDownTime: 0,
+        allowClick: false,
+      };
+    },
+    computed: {
+      coreOutlineFocus() {
+        return {
+          ':focus': {
+            ...this.$coreOutline,
+          },
+        };
+      },
+      headerLevel() {
+        return 'h' + this.headingLevel;
+      },
+      titleStyle() {
+        return {
+          color: this.$themeTokens.text,
+          fontSize: '16px',
+          margin: '0px 16px',
+        };
+      },
+      cardStyle() {
+        return {
+          backgroundColor: this.$themeTokens.surface,
+        };
+      },
+    },
+    methods: {
+      cardFocus(e) {
+        this.$emit('focus', e);
+      },
+      cardHover(e) {
+        this.$emit('hover', e);
+      },
+      cardClick() {
+        this.$router.push(this.to);
+      },
+      onMouseDown() {
+        this.mouseDownTime = new Date().getTime();
+      },
+      onMouseUp() {
+        const mouseUpTime = new Date().getTime();
+        if (mouseUpTime - this.mouseDownTime < 200) {
```

Comment:
I think it would be nice to have a comment here explaining that the reason for this behaviour is to allow selecting text inside the card.